test/swe2d/test_anisotropic.py

```diff
@@ -30,83 +50,107 @@ def run_steady_turbine(**model_options):
     options.timestepper_options.solver_parameters = {
         'mat_type': 'aij',
         'snes_type': 'newtonls',
+        'snes_linesearch_type': 'bt',
         'snes_rtol': 1e-8,
+        'snes_max_it': 100,
         'snes_monitor': None,
+        'snes_converged_reason': None,
+        'ksp_type': 'preonly',
+        'ksp_converged_reason': None,
         'pc_type': 'lu',
         'pc_factor_mat_solver_type': 'mumps',
     }
     options.output_directory = 'outputs'
     options.fields_to_export = ['uv_2d', 'elev_2d']
     options.use_grad_div_viscosity_term = False
-    options.element_family = 'dg-dg'
-    options.horizontal_viscosity = Constant(1.0)
-    options.quadratic_drag_coefficient = Constant(0.0025)
+    options.element_family = 'dg-cg'
+    options.horizontal_viscosity = viscosity
+    options.quadratic_drag_coefficient = drag_coefficient
     options.use_lax_friedrichs_velocity = True
     options.lax_friedrichs_velocity_scaling_factor = Constant(1.0)
     options.use_grad_depth_viscosity_term = False
     options.update(model_options)
     solver_obj.create_equations()
 
-    # Apply boundary conditions
+    # apply boundary conditions
     solver_obj.bnd_functions['shallow_water'] = {
-        1: {'uv': Constant([3.0, 0.0])},
-        2: {'elev': Constant(0.0)},
-        3: {'un': Constant(0.0)},
+        1: {'uv': inflow_velocity},  # inflow condition used upstream
+        2: {'elev': Constant(0.0)},  # Dirichlet condition at outflow to close system
```

Comment:
The elev boundary condition is imposed weakly, so it's not really a Dirichlet condition.